### ISSUE TITLE
cannot clone testcase

### ISSUE THREAD

**Issue by radiateboy** [kiwitcms/Kiwi#695]
### Steps to Reproduce

1. open case 
2. click clone button 
3. filter plan , don't choose plan
![20190107181407](https://user-images.githubusercontent.com/9248754/50762517-1e14dc00-12a8-11e9-826b-16ddd92a1972.png)
![20190107181429](https://user-images.githubusercontent.com/9248754/50762525-210fcc80-12a8-11e9-95b3-aeabfaa024ac.png)

**Comment by atodorov** [comment]
> filter plan , don't choose plan

Well you've got to select a TestPlan in which the TestCase will be cloned. You've got an error message telling you that this field is required.

However I'm able to reproduce a problem in which when you press Filter the results are shown on a different screen. 

I can reproduce on the demo website but not locally with master. I'm leaving this open for now to re-test with the new version which will be released very soon.

**Comment by asankov** [comment]
Will start to work on this once #683 is merged

**Comment by atodorov** [comment]
> Will start to work on this once #683 is merged

Stop working on #683 for now. I was able to reproduce locally with master:

1) Open a stand alone TestCase (e.g. not from a test plan) or edit the URL and remove the `?from_plan=xxx` parameter
2) Click CLONE
3) Then click Filter to filter the existing test plans in which to clone the new TC

Results:
A blank page shows with a list of test plans.

Expected:
The list of filtered plans should be shown in the same page below the filtering form so that a new plan can be selected and the clone operation can proceed.

Additional info:
This is not reproducing when the `?from_plan=xx` parameter is present.

**Comment by atodorov** [comment]
@radiateboy see the steps to reproduce above. Based on that you can workaround the problem by opening the test cases you want to clone from their test plan.

**Comment by atodorov** [comment]
@asankov see what binds this to the form. Maybe in one of the cases that is simply not initialized.

**Comment by asankov** [comment]
@atodorov you were right. in the case we have are missing `from_plan` the jQuery binder is never called. why is that, here is my explanation:
this is the javascript we are executing when the page is loaded
```javascript
$('#id_product').change(update_version_select_from_product);
if (!$('#id_version').val().length) {
    update_version_select_from_product();
}

jQ('#id_form_search_plan').bind('submit', function(e) {
  // logic we need for the form to work as expected
});
```
`update_version_select_from_product` gets the product and based on it, loads the product versions. after that it populates the `select` with the given versions and calls `$(selector).selectpicker('refresh')` However, the last line throws an error as `bootstrap-select` is never imported.

The valid question here is: Why does this work when we have a product? The answer is:
loading the product versions is achieved by an an asynchronous JSON-RPC call that takes time. while we wait for the call to happen we continue to run the main js file and therefore the logic we need for the form to work is executed. thus the code is executed in the following order:
```javascript
$('#id_product').change(update_version_select_from_product);
if (!$('#id_version').val().length) {
    //asynchronous code
    jsonRPC('Version.filter', {product: product_id}, updateVersionSelectCallback);
}

jQ('#id_form_search_plan').bind('submit', function(e) {
  // logic we need for the form to work as expected
});

// here somewhere the asynchronous code is executed
function updateVersionSelectCallback(products) {
  // code
  $(selector).selectpicker('refresh'); // -> error. nothing more is executed
}
```

when product is missing however, we don't do an aynschronous call, and load empty versions. therefore the code is executed in another order:
```javascript
$('#id_product').change(update_version_select_from_product);
if (!$('#id_version').val().length) {
    //synchronous code
     $(selector).selectpicker('refresh'); // -> error. nothing more is executed
}

// none of this is executed
jQ('#id_form_search_plan').bind('submit', function(e) {
  // logic we need for the form to work as expected
});
}
```

This is easily fixed by importing `bootstrap-select` in the template. However, then arises another problem. This breaks the layout and the functionality. It looks like this:
<img width="470" alt="screenshot 2019-01-08 at 22 54 52" src="https://user-images.githubusercontent.com/18421997/50858212-74614800-1398-11e9-885a-8a233d37ffa7.png">

I suppose that `bootstrap-switch` and templates with Django form field, e.g. `                                            {{ search_form.version }}` are not compatible and in order for this to work, we need to refactor the tempate to Patternfly where we don't use Django form fields, or remove the `bootstrap-switch`.
